ts/features/bonus/bpd/store/reducers/details/transactionsv2/entities.ts

```diff
@@ -0,0 +1,184 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import { getType } from "typesafe-actions";
+import { WinningTransactionMilestoneResource } from "../../../../../../../../definitions/bpd/winning_transactions_v2/WinningTransactionMilestoneResource";
+import { WinningTransactionPageResource } from "../../../../../../../../definitions/bpd/winning_transactions_v2/WinningTransactionPageResource";
+import { Action } from "../../../../../../../store/actions/types";
+import {
+  IndexedById,
+  toArray,
+  toIndexed
+} from "../../../../../../../store/helpers/indexer";
+import { AwardPeriodId } from "../../../actions/periods";
+import {
+  BpdTransactionId,
+  bpdTransactionsLoadMilestone,
+  bpdTransactionsLoadPage,
+  BpdTransactionV2
+} from "../../../actions/transactions";
+
+export type BpdPivotTransaction = {
+  idTrx: BpdTransactionId;
+  amount: number;
+};
+
+export type BpdTransactionsEntityState = {
+  pivot: pot.Pot<BpdPivotTransaction | null, Error>;
+  byId: IndexedById<BpdTransactionV2>;
+  foundPivot: boolean;
+};
+type NormalizedTransactions = {
+  data: ReadonlyArray<BpdTransactionV2>;
+  found: boolean;
+};
+
+const initState: BpdTransactionsEntityState = {
+  pivot: pot.none,
+  foundPivot: false,
+  byId: {}
+};
+
+const getPeriodEntry = (
+  state: IndexedById<BpdTransactionsEntityState>,
+  id: AwardPeriodId
+): BpdTransactionsEntityState => state[id] ?? initState;
+
+const updatePeriodEntry = (
+  input: IndexedById<BpdTransactionsEntityState>,
+  period: AwardPeriodId,
+  newVal: BpdTransactionsEntityState
+): IndexedById<BpdTransactionsEntityState> => ({
+  ...input,
+  [period]: newVal
+});
+
+/**
+ * Normalize the cashback amount for the transactions, using as ref the pivot transaction
+ * - pivot === null, the user did not reach the max cashback amount and all the transactions are valid
+ * - pivot !== null && not found && idTrx === pivot.id, pivot transaction found, all the following transaction have the complete value
+ * - pivot !== null && found, the cashback transaction has the complete value
+ * @param transactions
+ * @param pivot
+ * @param foundPivot
+ */
+const normalizeCashback = (
+  transactions: ReadonlyArray<WinningTransactionMilestoneResource>,
+  pivot: BpdPivotTransaction | null,
+  foundPivot: boolean
+): NormalizedTransactions => {
+  // eslint-disable-next-line functional/no-let
+  let found = foundPivot;
+  return {
+    data: transactions.map(x => {
+      if (found || pivot === null) {
+        return { ...x, validForCashback: true } as BpdTransactionV2;
+      }
+      if (x.idTrx === pivot.idTrx) {
+        found = true;
+        return {
+          ...x,
+          cashback: pivot.amount,
+          validForCashback: true
+        } as BpdTransactionV2;
+      }
+
+      return { ...x, cashback: 0, validForCashback: false } as BpdTransactionV2;
+    }),
+    found
+  };
+};
+
+const updateTransactions = (
```

Comment:
Added comment in 43d99baf308c8f0bcbd7a72bdc72f7d2e813fde2 & 0ce62bc121e15ce72503168133ff53a985b173ab. 
Yes, in order to avoid to recalculate the normalization for all the existing transactions foreach new page, I keep the state using `foundPivot`; in this way only the new pages (delta data) should be normalized.